### PR TITLE
Gapaniuk HW12i13 and CW15 and CW16

### PR DESCRIPTION
**Pull request by GapSerg**
HW12i13

### CODE REVIEW

File: src/main/java/edu/academy/jc/gapaniuk/hw12i13/SumAccumulator.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        this.sumList = list;
+    }
+
+    public int sum = 0;
+
+    @Override
+    public void run() {
+        int s1;
+        int i = 0;
+        synchronized (this.sumList) {
```
Comment (StanislavLapitsky) [inline src/main/java/edu/academy/jc/gapaniuk/hw12i13/SumAccumulator.java:26]: Actually here we also should care about synchronized block. The same as in Generator

Comment (GapSerg) [reply src/main/java/edu/academy/jc/gapaniuk/hw12i13/SumAccumulator.java:26]: Sorry, but I don't anderstend you?

File: src/main/java/edu/academy/jc/gapaniuk/cw15/FirstTask.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+public class FirstTask {
+    public static void main(String[] args) {
+        Set<String> count = Stream
+                .generate(() -> new Random().nextInt(10))
+                .limit(10)
+                .filter(intValue -> {
+                    return intValue % 2 == 0;
+                })
+                .map(intValue -> intValue.toString())
+                .collect(Collectors.toSet()); //завершающая операция
```
Comment (StanislavLapitsky) [inline src/main/java/edu/academy/jc/gapaniuk/cw15/FirstTask.java:17]: Please don't use comments in Russian

File: src/main/java/edu/academy/jc/gapaniuk/cw15/SecondTask.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.stream.Stream;
+
+public class SecondTask {
+    public static void main(String[] args) {
+
+
+        int n = 50;
+        Integer a = Stream
+                .iterate(2, x -> (x * 5) < n, x -> x + 2)
+                //.peek(System.out::println)
```
Comment (StanislavLapitsky) [inline src/main/java/edu/academy/jc/gapaniuk/cw15/SecondTask.java:13]: Please remove the commented code

File: src/main/java/edu/academy/jc/gapaniuk/cw15/ThirdTask.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package edu.academy.jc.gapaniuk.cw15;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ThirdTask {
+    public static void main(String[] args) {
+        List<String> numbers = Arrays.asList("1", "2", "3", "4", "5","6");
+        StringBuilder sB = new StringBuilder();
+//        StringBuilder sumArray = numbers
```
Comment (StanislavLapitsky) [inline src/main/java/edu/academy/jc/gapaniuk/cw15/ThirdTask.java:10]: Looks like no logic at all. The commented code to be removed. Actual logic to be supported

**Review by StanislavLapitsky — APPROVED** [review]

**Pull request merged**